cratedb_toolkit/testing/testcontainers/cratedb.py

```diff
@@ -101,3 +150,53 @@ def _connect(self):
         #       In `testcontainers-java`, there is the `HttpWaitStrategy`.
         # TODO: Provide a client instance.
         wait_for_logs(self, predicate="o.e.n.Node.*started", timeout=MAX_TRIES)
+
+
+class CrateDBFixture:
+    """
+    A little helper wrapping Testcontainer's `CrateDBContainer` and
+    CrateDB Toolkit's `DatabaseAdapter`, agnostic of the test framework.
+    """
+
+    def __init__(self, crate_version: str = "nightly", **kwargs):
+        self.cratedb: Optional[CrateDBContainer] = None
+        self.database: Optional[DatabaseAdapter] = None
+        self.image: str = "crate/crate:{}".format(crate_version)
+
+    def start(self, **kwargs):
+        """
+        Start testcontainer, used for tests set up
+        """
+        self.cratedb = CrateDBContainer(image=self.image, **kwargs)
+        self.cratedb.start()
+        self.database = DatabaseAdapter(dburi=self.get_connection_url())
+
+    def stop(self):
+        """
+        Stop testcontainer, used for tests tear down
+        """
+        if self.cratedb:
+            self.cratedb.stop()
+
+    def reset(self, tables: Optional[list] = None):
+        """
+        Drop tables from the given list, used for tests set up or tear down
+        """
+        if tables and self.database:
+            for reset_table in tables:
+                self.database.connection.exec_driver_sql(f"DROP TABLE IF EXISTS {reset_table};")
+
+    def get_connection_url(self, *args, **kwargs):
+        """
+        Return a URL for SQLAlchemy DB engine
+        """
+        if self.cratedb:
+            return self.cratedb.get_connection_url(*args, **kwargs)
+        return None
+
+    @property
+    def http_url(self):
+        """
+        Return a URL for HTTP interface
+        """
+        return self.get_connection_url(dialect="http")
```

Comment:
#### Re.: Naming things

`get_connection_url` and `http_url` may also be renamed. However, it is also not relevant for this patch, but merely meant to encourage you that you can easily come up with ideas for renaming, because `cratedb-toolkit` is very young, and most of its code didn't see more than two or three iterations yet, so it offers many opportunities in this regard.

Reply:
When naming these, we can go in two directions:
1. rely on the `testcontainers` naming conventions, e.g. if it's a connection URL that we get using `testcontainers.core.generic.DbContainer._create_connection_url` and the interface sugests to implement `get_connection_url`, then `get_connection_url` seems very natural to me.
2. rely on `CrateDB` naming. In this case we can say it's getting the [host](https://cratedb.com/docs/crate/crash/en/latest/run.html) url or the [interface](https://cratedb.com/docs/crate/reference/en/5.5/interfaces/index.html#) url. So that we can rename `get_connection_url` to `get_interface_url` or `get_host_url`

Tbh, I think the current naming is ok. If someone needs to go deeper into `testcontainers` base classes, `get_connection_url` looks better to me.

Reply:
I'd rename `http_url` though to `get_http_url` to make it consistent with the `get_connection_url`.

Reply:
I agree with your assessment to follow the naming conventions of Testcontainers. Maybe we can have an inline docstring comment on the `get_connection_url` method to tell the readers that it is actually an SQLAlchemy database connection uri that will be returned here?

Reply:
`get_connection_url` docstring says: `Return a URL for SQLAlchemy DB engine`. Do you want it to put it differently anyhow?

Reply:
Ah. No it is perfect. Apologies if I missed it.